src/mkdocs_autorefs/plugin.py

```diff
@@ -69,13 +84,22 @@ def get_item_url(self, anchor: str) -> str:
             KeyError: If there isn't an item by this identifier anywhere on the site.
         """
         try:
-            return self._url_map[anchor]
+            url = self._url_map[identifier]
         except KeyError:
-            new_anchor = self.get_fallback_anchor(anchor)
-            if new_anchor and new_anchor in self._url_map:
-                return self._url_map[new_anchor]
+            if identifier in self._abs_url_map:
+                return self._abs_url_map[identifier]
+
+            if fallback:
+                new_identifier = fallback(identifier)
+                if new_identifier:
+                    return self.get_item_url(new_identifier, from_url)
```

Comment:
Don't we risk infinite recursion here?

Reply:
Oh, sorry that there's no explanation for the "cleverness".

The self-call is not exactly the same, `fallback` is dropped. So, no, there will be only 2 attempts.

I'll add an explicit `fallback=None` for clarity.

Reply:
Ah, my bad for not reading this more thoroughly, I would have seen it :slightly_smiling_face: 
Thanks for the explanation. The `fallback=None` is not necessary, but you can add it if you want.

Reply:
And you did. Thanks again ^^

Reply:
Ooh let me revert that, there's some failure, and it's a concern for backwards compatibility.
I'll also check in detail why the test failed, I would not have expected that

Reply:
This is resolved btw. There wasn't really a problem, just an implicit merge conflict